ast/src/main/java/com/graphicsfuzz/common/ast/decl/ArrayInfo.java

```diff
@@ -17,37 +17,58 @@
 package com.graphicsfuzz.common.ast.decl;
 
 import com.graphicsfuzz.common.ast.IAstNode;
+import com.graphicsfuzz.common.ast.expr.Expr;
+import com.graphicsfuzz.common.ast.expr.IntConstantExpr;
 import com.graphicsfuzz.common.ast.visitors.IAstVisitor;
+import com.graphicsfuzz.common.ast.visitors.UnsupportedLanguageFeatureException;
 import java.util.Optional;
 
 public class ArrayInfo implements IAstNode {
 
-  private Optional<Integer> size;
+  private Optional<Expr> size;
+  private Optional<Expr> originalSize;
 
-  public ArrayInfo(Integer size) {
+  public ArrayInfo(Expr size) {
     this.size = Optional.of(size);
+    this.originalSize = Optional.of(size);
   }
 
   public ArrayInfo() {
     this.size = Optional.empty();
+    this.originalSize = Optional.empty();
   }
 
   public boolean hasSize() {
     return size.isPresent();
   }
 
```

Comment:
Let's rename "getSize" to "getConstantSize" to make it clearer that it is not a plain getter method.

Can you add a JavaDoc comment for this method?  (We try to do that for new code.)